nncf/experimental/torch/fx/transformations.py

```diff
@@ -170,51 +169,63 @@ def insert_one_qdq_after_node(model: torch.fx.GraphModule, target_node: torch.fx
                 # for qparams that are not scale/zero_point (like axis, dtype) we store
                 # them as literals in the graph.
                 quantize_op_inputs.append(value_or_node)
-    with graph.inserting_after(target_node):
+
+    _, feed_node, ctx_manager = _get_data_for_insertion(graph, target_point)
+    quantize_op_inputs[0] = feed_node
+    with ctx_manager(target_node):
         quantized_node = graph.create_node(node_type, quantize_op, tuple(quantize_op_inputs), {})
-        # use the same qparams from quantize op
+
+    # use the same qparams from quantize op
     dq_inputs = [quantized_node] + quantize_op_inputs[1:]
-    user_dq_nodes = []
-    with graph.inserting_after(quantized_node):
-        for user in target_node.users:
-            if user is quantized_node:
-                continue
-            user_dq_nodes.append((user, graph.call_function(dequantize_op, tuple(dq_inputs), {})))
+    if target_point.target_type == TargetType.OPERATOR_POST_HOOK:
```

Comment:
TargetType contains 8 values, if-else is not correct way to check type.
Add check that target_type contains only expected values, or use if-elif-else-error.

And looks that  reusing PTTargetPoint and PTTargetType is bad idea, contain too many unrelated code.

Reply:
Done